libs/media-player/src/audio-annotate.cy.tsx

```diff
@@ -0,0 +1,234 @@
+    const validAudioUrl =
+        'https://coscrad.org/wp-content/uploads/2023/05/metal-mondays-mock2_370934__karolist__guitar-solo.mp3';
+
+    const validShorterDurationAudioUrl =
+        'https://coscrad.org/wp-content/uploads/2023/05/metal-mondays-mock1_533847__tosha73__distortion-guitar-power-chord-e.wav';
+
+    describe(`when there is a single, fixed value for audioUrl`, () => {
+        beforeEach(() => {
+            cy.mount(<AudioAnnotatorWidget audioUrl={validAudioUrl} />);
+        });
+
+        it('should display audio controls', () => {
+            cy.get('audio').should('be.visible');
+            cy.get('audio').should('have.prop', 'controls', true);
+        });
+
+        describe(`when the media currentTime is 0 the mark in- and out-point buttons`, () => {
+            it(`should be disabled`, () => {
+                cy.getByDataAttribute('in-point-marker-button').should('be.disabled');
+                cy.getByDataAttribute('out-point-marker-button').should('be.disabled');
+            });
+        });
+
+        describe(`when the media has been played`, () => {
+            beforeEach(() => {
+                cy.get('audio').then(([audioElement]) => {
+                    audioElement.play();
+                });
+            });
+
+            describe(`the mark in-point button`, () => {
+                it(`should be enabled`, () => {
+                    cy.getByDataAttribute('in-point-marker-button').should('be.enabled');
+                });
+            });
+
+            describe(`the mark out-point button`, () => {
+                it(`should be disabled`, () => {
+                    cy.getByDataAttribute('in-point-marker-button').should('be.disabled');
+                });
+            });
+
+            describe(`when the mark in-point button is clicked`, () => {
+                it(`should mark an in-point`, () => {
+                    cy.wait(2000);
+
+                    cy.getByDataAttribute('in-point-marker-button').click();
+                });
+            });
+
+            describe(`when the mark in-point button is clicked, the mark out-point button`, () => {
+                it('should be enabled', () => {
+                    cy.wait(2000);
+
+                    cy.getByDataAttribute('in-point-marker-button').click();
+
+                    cy.wait(2300);
+
+                    cy.getByDataAttribute('out-point-marker-button').should('be.enabled');
+                });
+            });
+
+            describe(`when the mark in-point button is clicked, clicking the mark out-point button`, () => {
+                it('should mark an out-point', () => {
+                    cy.wait(2000);
+
+                    cy.getByDataAttribute('in-point-marker-button').click();
+
+                    cy.wait(2500);
+
+                    cy.getByDataAttribute('out-point-marker-button').click();
+
+                    cy.getByDataAttribute('time-range-selected').should('exist');
```

Comment:
The data attribute here sounds like an event. Rename it to `selected-time-range` or `time-range-info` for readability.